### ISSUE TITLE
Trailing comma on single-line nested dict

### ISSUE THREAD

**Issue by NakulK48** [psf/black#1465]
**Describe the bug** A clear and concise description of what the bug is.

If a dict is nested within another dict, it doesn't reformat correctly.

**To Reproduce** Steps to reproduce the behavior:

Take this:

```python
my_dict = {
    "outer": {
        "a": 1,
        "b": 2,
    }
}
```

Pass it through black:

```python
my_dict = {"outer": {"a": 1, "b": 2,}}
```

This has a trailing comma, which violates https://www.flake8rules.com/rules/E231.html

**Expected behavior** A clear and concise description of what you expected to happen.

Either the snippet unchanged (which is what happens when the dictionary isn't nested)

```python
my_dict = {
    "a": 1,
    "b": 2,
}
```

Or a single-line dict with no trailing comma (`black` on this returns with no change)

```python
my_dict = {"outer": {"a": 1, "b": 2}}
```

**Environment (please complete the following information):**

- Version: 19.10b0
- OS and Python version: Ubuntu 18.04, python3.8.2

**Does this bug also happen on master?**

Yes

**Additional context** Add any other context about the problem here.

**Comment by NakulK48** [comment]
I'll add as a stylistic note that it's kind of nice to be able to enforce dict fields going on new lines without having to do `fmt: off`

**Comment by ichard26** [comment]
This is part of #1288. The 'Magic Trailing Comma' doesn't work on nested collections yet.

**Comment by ichard26** [comment]
Fixed by #1612 so the trailing comma now works on nested collections.

```console
(black-rHKUX7ap) R:\Programming\black>type temp.py
my_dict = {
    "outer": {
        "a": 1,
        "b": 2,
    },
}

(black-rHKUX7ap) R:\Programming\black>black temp.py --diff --color
All done! ✨ 🍰 ✨
1 file would be left unchanged.
``` 

**Environment:**
* Black version: _Black_ `20.08b1`
* Python version: `CPython 3.8.1`
* OS version: `Windows 10 Home Edition Build 18363`
